### ISSUE TITLE
Repeated invocations of CSL.makeAdhocBibliography() crashes JVM

### ISSUE THREAD

**Issue by nhnb** [michel-kraemer/citeproc-java#79]
## 🐞 Description

`makeAdhocBibliography()` provides an easy way to create a bibliography. The documentation warns that it is not the best approach to call this method repeatedly for performance reasons.

The issue, however, is much worse: The JVM crashes with an native memory allocation failure in C code at less than 2000 invocations.

## 🔥 Error messages

~~~~
# Fatal error in CALL_AND_RETRY_LAST
# Allocation failed - process out of memory

# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGILL (0x4) at pc=0x00007fce21cb0a19, pid=9855, tid=0x00007fce44158700
# JRE version: OpenJDK Runtime Environment (8.0_222-b10) (build 1.8.0_222-8u222-b10-1ubuntu1~18.04.1-b10)
# Java VM: OpenJDK 64-Bit Server VM (25.222-b10 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libj2v8_linux_x86_64.so+0xae3a19]  v8::base::OS::Abort()+0x9
~~~~

## 📝 How to reproduce

Make sure that V8 is on the classpath and execute the following java program.

```java
import java.io.IOException;

import de.undercouch.citeproc.CSL;
import de.undercouch.citeproc.csl.CSLItemDataBuilder;
import de.undercouch.citeproc.csl.CSLNameBuilder;
import de.undercouch.citeproc.csl.CSLType;
import de.undercouch.citeproc.output.Bibliography;

public class Test {

    public static void main(String[] args) throws IOException {
        CSLItemDataBuilder builder = new CSLItemDataBuilder();

        builder.type(CSLType.ARTICLE)
                .title("Titel")
                .issued(2005)
                .author(new CSLNameBuilder().given("Firstname").family("Lastname").build())
                .abstrct("Abstract")
                .volume("1")
                .issue("1")
                .page("1-2");

        long start = System.currentTimeMillis();

        for (int i = 0; i< 1000000; i++) {
            Bibliography bib = CSL.makeAdhocBibliography("apa-old-doi-prefix", "html", builder.build());
            @SuppressWarnings("unused")
            String ignore = bib.getEntries()[0];

            if (i % 100 == 0) {
                System.out.println(i + "\t" + (System.currentTimeMillis() - start));
            }
        }
    }

}
```

## 🌍  Environment
- JRE version: OpenJDK Runtime Environment (8.0_222-b10) (build 1.8.0_222-8u222-b10-1ubuntu1~18.04.1-b10)
- Java VM: OpenJDK 64-Bit Server VM (25.222-b10 mixed mode linux-amd64 compressed oops)
- V8 JavaScript Engine

## 🔭 Additional information

As far as JVM memory is concerned, the GC seems to be working fine:

![jvm-memory](https://user-images.githubusercontent.com/364184/69238738-00d3be00-0b99-11ea-976d-6c59ab0efcdf.png)

**Comment by nhnb** [comment]
[hs_err_pid9855.log](https://github.com/michel-kraemer/citeproc-java/files/3869205/hs_err_pid9855.log)

**Comment by nhnb** [comment]
Oh, nice. **You fixed the issue in 2.0.0**. I should have checked for a new version beforehand.


Anyway, I can confirm, that calling `makeAdhocBibliography()` 50000 times works perfectly fine with 2.0.0: The time for each block of 100 invocations increases only very slow. JVM-memory consumption stays below 50 MB. RSS is only increasing slowly.

![time](https://user-images.githubusercontent.com/364184/69245455-1f40b600-0ba7-11ea-9be0-161e49417887.png)

![jvm-memory_2 0 0](https://user-images.githubusercontent.com/364184/69245288-db4db100-0ba6-11ea-9174-a2b76d0985b9.png)

**Comment by michel-kraemer** [comment]
Yes. I think it was this commit: 27ee6b738b4aaa6bbde1d59e82706b33720a8bfd Anyhow, thanks for the thorough testing!
